graphene_parser.py

```diff
@@ -325,10 +324,24 @@ def parse_file(
         self._current_file = None
         self._include_hierarchy = None
 
-    def get_function_body_trees(
+    def get_functions_to_codegen(
         self,
-    ) -> list[tuple[SourceLocation, cg.Function, Tree, cg.GenericMapping]]:
-        return self._function_body_trees
+    ) -> Generator[
+        tuple[cg.FnDeclaration, cg.FunctionSignature, cg.GenericMapping, Tree],
+        None,
+        None,
+    ]:
+        while to_generate := self._program.symbol_table.get_next_function_to_codegen():
```

Comment:
Before we sometimes codegened functions with types that the signature disallowed (because we were doing it all with blind callbacks), this doesn't happen anymore. However, we will still sometimes generate extra functions eg. if the pattern match + SFINAE succeeds, the function will get codegened. It may not end up being used because there might be another function with a lower type conversion cost

Reply:
That's not too bad I suppose. We can still do better but I wouldn't bother right now